Packs/AWS-WAF/Integrations/AWS-WAF/AWS_WAF.py

```diff
@@ -0,0 +1,1122 @@
+        regex_set_arn: The regex set to match to
+
+    Returns:
+        A regex match statement value
+    """
+    web_request_component_object = build_web_component_match_object(web_request_component, oversize_handling)
+    return {
+        'ARN': regex_set_arn,
+        'FieldToMatch': {
+            web_request_component: web_request_component_object
+        },
+        'TextTransformations': [
+            {'Priority': 0, 'Type': text_transformation
+             }
+        ]
+    }
+
+
+def build_string_match_rule_object(args: dict) -> dict:
+    """
+    Creates a string match rule statement object that can be added to a rule
+    Args:
+        args: The command arguments
+
+    Returns:
+        String match rule statement object
+    """
+    string_match_statement: dict = {
+        'Statement': build_string_match_statement(args)
+    }
+    return string_match_statement
+
+
+def build_new_rule_object(args: dict, rule_group_visibility_config: dict,
+                          build_rule_func: Callable[[dict], dict]) -> dict:
+    """
+    Creates a country rule object that can be added to a rule group rules list
+    Args:
+        args: The command arguments
+        rule_group_visibility_config: The rule visibility config
+        build_rule_func: A generic function that builds the statement of the rule
+
+    Returns:
+        Entire rule object
+    """
+    name = args.get('rule_name', '')
+    rule_visibility_config = build_visibility_config_object(
+        metric_name=name,
+        cloud_watch_metrics_enabled=rule_group_visibility_config.get('CloudWatchMetricsEnabled'),  # type: ignore
+        sampled_requests_enabled=rule_group_visibility_config.get('SampledRequestsEnabled'))  # type: ignore
+
+    rule = {
+        'Name': name,
+        'Priority': arg_to_number(args.get('priority', '')) or 0,
+        'Action': {
+            args.get('action'): {}
+        },
+        'VisibilityConfig': rule_visibility_config,
+
+    }
+    rule |= build_rule_func(args)
+
+    return rule
+
+
+def delete_rule(rule_name: str, rules: list) -> list:
+    """
+    Deletes a rule from a rules list
+    Args:
+        rule_name: The rule name to delete
+        rules: The rules list of a rule group
+
+    Returns:
+        A new rules list without the rule to delete
+    """
+    updated_rules = rules.copy()
+    for rule in rules:
+        if rule.get('Name') == rule_name:
+            updated_rules.remove(rule)
+            break
+    return updated_rules
+
+
+def append_new_rule(rules: list, rule: dict) -> list:
+    """
+    Adds a rule from a rules list
+    Args:
+        rule: The rule object to add
+        rules: The rules list of a rule group
+
+    Returns:
+        A new rules list with the rule to delete
+    """
+    updated_rules = rules.copy()
+    updated_rules.append(rule)
+    return updated_rules
+
+
+def build_kwargs(args: dict) -> dict:
```

Comment:
is it used in all the commands for building the request?

Reply:
No, I'll change the name so it will be clearer